Document RKE2 cluster provisioning, node pools, Windows features (#3476)
rancher/docs: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/node-pools/_index.md

# Node Drivers

If you don't find the node driver that you want to use, you can see if it is available in Rancher's built-in [node drivers and activate it]({{<baseurl>}}/rancher/v2.6/en/admin-settings/drivers/node-drivers/#activating-deactivating-node-drivers), or you can [add your own custom node driver]({{<baseurl>}}/rancher/v2.6/en/admin-settings/drivers/node-drivers/#adding-custom-node-drivers).

# Node Roles in RKE2

> The RKE2 CLI has two roles, `server` and `agent`.  RKE2 CLI was designed not to expose node roles as a first-class concept. With RKE2 integration in Rancher v2.6, RKE2 node pools can use `etcd`, `controlplane` or `worker` roles within Rancher.

Reviewer: I do not have a great alternative formulation but this reads a bit awkward to me. Maybe something like:
> The RKE2 CLI exposes two roles, `server` and `agent`, which represent the k8s node-roles `etcd`+`control-plane` and `worker` respectively. With RKE2 integration in Rancher v2.6, RKE2 node pools can represent more fine-grained role assignments such that `etcd` and `control-plane` roles can be separated.

Author: that's a much better way of phrasing it, thank you.

> The RKE2 CLI has two roles, `server` and `agent`.  RKE2 CLI was designed not to expose node roles as a first-class concept. After RKE2 was fully integrated into Rancher v2.6, RKE2 node pools can use `etcd`, `controlplane` or `worker` roles within Rancher.

Reviewer 2: ```suggestion
The RKE2 CLI has two roles, `server` and `agent`.  RKE2 CLI was designed not to expose node roles as a first-class concept. With RKE2 integration in Rancher v2.6, RKE2 node pools can use `etcd`, `controlplane` or `worker` roles within Rancher.
```

> The same functionality of using `etcd`, `controlplane` and `worker` nodes is possible in the RKE2 CLI by using flags and node tainting to control where workloads and the Kubernetes master were scheduled. The reason those roles were not implemented as first-class roles in the RKE CLI is that RKE2 is conceptualized as a set of raw building blocks that are best leveraged through an orchestration system such as Rancher.

Reviewer 2: ```suggestion
The same functionality of using `etcd`, `controlplane` and `worker` nodes is possible in the RKE2 CLI by using flags and node tainting to control where workloads and the Kubernetes master were scheduled. The reason those roles were not implemented as first-class roles in the RKE2 CLI is that RKE2 is conceptualized as a set of raw building blocks that are best leveraged through an orchestration system such as Rancher.
```
